src/hooks/useIsOnScreen.ts

```diff
@@ -34,7 +34,7 @@ function getScrollParent(node: HTMLElement | null): HTMLElement | typeof window
     return null;
   }
 
-  if (node.scrollHeight > node.clientHeight) {
+  if (node.scrollHeight > node.clientHeight && node.clientHeight > 0) {
```

Comment:
The ui-kit scroll container wraps things with a `0` height `0` width "scrollable" container (freaking weird lol), which breaks this behavior. Added this extra check to ignore 0 height parent containers, which fixes the issue (I yalc'd and confirmed this in platform).

<img width="1494" alt="Screen Shot 2021-12-02 at 4 42 23 PM" src="https://user-images.githubusercontent.com/847542/144507791-91a2305a-8190-4485-a3b9-fa1c2759d7f0.png">